packages/i18n/src/translator.ts

```diff
@@ -0,0 +1,104 @@
+import Mustache from 'mustache';
+import Catalog from './catalog';
+import en_US from './locales/en_US';
+
+/**
+ * The default locale.
+ */
+const DEFAULT_LOCALE = 'en_US';
+
+/**
+ * Locale mappings.
+ */
+const MAPPINGS = {
+  'en_US': en_US
+};
+
+/**
+ * The help template.
+ */
+const TEMPLATE = 'Description: {{description}}\n\n' +
+  'Link: {{link}}\n\n' +
+  'Example: {{example}}\n\n' +
+  'Returns: {{returns}}';
+
+/**
+ * The translator class.
+ */
+class Translator {
+  private locale: string = DEFAULT_LOCALE;
+  private catalog: Catalog;
+
+  /**
+   * Translate the key.
+   *
+   * @param {string} key - The key.
+   *
+   * @returns {string} The translation.
+   */
+  __(key: string): string {
+    return this.translate(key);
+  }
+
+  /**
+   * Instantiate the catalog.
+   *
+   * @param {Catalog} catalog - The optional catalog.
+   */
+  constructor(catalog?: Catalog) {
+    if (catalog) {
+      this.catalog = catalog;
+    } else {
+      this.catalog = MAPPINGS[this.locale];
+    }
+  }
+
+  /**
+   * Set the current locale.
+   *
+   * @param {string} locale - The locale.
+   */
+  setLocale(locale: string) {
+    if (MAPPINGS.hasOwnProperty(locale)) {
+      this.locale = locale;
+      this.catalog = MAPPINGS[locale];
+    } else {
+      this.locale = DEFAULT_LOCALE;
+      this.catalog = MAPPINGS[DEFAULT_LOCALE];
+    }
+  }
+
+  /**
+   * Get the translation for the provided key.
+   *
+   * @param {string} key - The key.
+   *
+   * @returns {string} The translation.
+   */
+  translate(key: string): string {
+    return this.find(key);
+  }
+
+  /**
+   * Translate the key into API help.
+   *
+   * @param {string} key - The key.
+   *
+   * @returns {string} The translated string.
+   */
+  translateApiHelp(key: string): string {
+    const value = this.find(key);
+    if (typeof value === 'string') {
+      return value;
+    }
+    return Mustache.render(TEMPLATE, value);
```

Comment:
I am wondering if we should let the implementations handle the formatting that this `TEMPLATE` is doing. Instead have `Translator` return just the translation object.

I am thinking we will run into trouble, for example, when we want to pad things in the browser vs. the shell, or add new extra new lines, or change colour of items.